### DISCUSSION TITLE
Preferred method for easy project switching

### DISCUSSION THREAD

**Discussion by jlmcgehee21** [ankitpokhrel/jira-cli#529] [Q&A] [answered]
Hello!

Thank you for the amazing tool!  I'm looking for the best supported method for context-aware projects and/or instance switching _without_ passing the `-c path/to/config` or the `-p` arg every time.  (would be great if it worked with a tool like direnv)

Does the tools support a custom config search path, so that I could embed a config file in a given directory, and it would search there _first_ for a config file, before using the default one?

If not? Any other methods?

Thank you!

**Comment by jlmcgehee21** [comment, marked as answer]
Update: 

I did find a workable method, adding the following to a `.envrc` file in each directory, which actively overwrites the config based on my directory context.

`jira config --force --installation cloud --server <url> --login <email> --project <project code> --board <board name>`

**Reply by ankitpokhrel** [reply]
> Does the tools support a custom config search path, so that I could embed a config file in a given directory, and it would search there first for a config file, before using the default one?

You can use `XDG_CONFIG_HOME` env to set custom config dir. The solution you posted above works better for your usecase I think.
